### PR TITLE
QField 💖 QML & HTML

### PR DESCRIPTION
**Pull request by nirvn**
This PR implements QML widget support in the feature form. An obligatory GIF:
Peek 2020-08-17 13-33

As the above screencast demonstrates, the current implementation QField supports expression.evaluate(...). The method I've settled on to do it (i.e. string substitution in the QML source string) was the most appropriate solution I could come up with. On the plus side, the general idea behind it can easily be used to come up with an HTML widget support (via the WebView item).

### CODE REVIEW

**Comment by nirvn** [comment]
BTW, the QML widget in the GIF nicely occupies the available width in QField:
Peek 2020-08-17 13-41

The way you set this up is by using the following declaration in the QML widget source code:
```
ChartView {
   ...
    width:parent.parent === null ? 400 : parent.width
   ...
}
```

Users can't simply use parent.width as in QGIS, the parent is a QQuickRootItem, which doesn't have a width value. Since there's no proper way to check whether a QML item is a root or not, we can check whether the parent has a parent (if it doesn't, it's the root).

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-1d16c3185ffdc68976a943ff21221f8a252f94ad-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-1d16c3185ffdc68976a943ff21221f8a252f94ad-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-1d16c3185ffdc68976a943ff21221f8a252f94ad-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-1d16c3185ffdc68976a943ff21221f8a252f94ad-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-bae291d43115c6206f76e2cec58345fe1574aef9-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-bae291d43115c6206f76e2cec58345fe1574aef9-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-bae291d43115c6206f76e2cec58345fe1574aef9-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-bae291d43115c6206f76e2cec58345fe1574aef9-x86_64.apk)

File: src/core/attributeformmodelbase.cpp
Comment (3nids) [inline src/core/attributeformmodelbase.cpp:246]: rather than manually looping in the text, I would rather use a regex: 
https://stackoverflow.com/a/19863847/1548052

You have to first remove the heading "expression.evaluate" and feed the regex with the remaining text:
https://regex101.com/r/eBtSTM/129

This regex (or your code) is not 100% robust as you could have unbalanced parenthesis in some strings (let's say a list of items: "a) … b)… c) …" ). But it might be acceptable.

File: src/core/featuremodel.cpp
Comment (3nids) [inline src/core/featuremodel.cpp:52]: why did you remove this?

File: src/core/attributeformmodelbase.cpp
Comment (nirvn) [reply src/core/attributeformmodelbase.cpp:246]: @3nids , I went down this road until I realized - while chatting with @m-kuhn  - that we can use an unescaped double quote as end delimiter. So yeah, regexp is the way forward. Modifying that now.

File: src/core/featuremodel.cpp
Comment (nirvn) [reply src/core/featuremodel.cpp:52]: To trigger a refresh when the feature is modified. If you look into FeatureModel::save(), you'll see it calls setFeature( modifiedFeature ), which ended up doing nothing as mFeature == modifiedFeature will be true even if the modifiedFeature has different attribute values et al.

Comment (3nids) [reply src/core/featuremodel.cpp:52]: is it worth adding a `bool force = false` argument to the method?

File: src/core/attributeformmodelbase.cpp
Comment (3nids) [reply src/core/attributeformmodelbase.cpp:246]: nice catch, cool

File: src/core/featuremodel.cpp
Comment (nirvn) [reply src/core/featuremodel.cpp:52]: Na, I've seen other weird behaviors which is fixed by this removed check. So I'd prefer just removing it all together.

Comment (m-kuhn) [reply src/core/featuremodel.cpp:52]: >  as mFeature == modifiedFeature will be true even if the modifiedFeature has different attribute values et al.

According to `QgsFeatore::operator==` attributes should be compared as well. What did I miss?

Comment (nirvn) [reply src/core/featuremodel.cpp:52]: I don't think so no.

Comment (3nids) [reply src/core/featuremodel.cpp:52]: https://github.com/qgis/QGIS/blob/master/src/core/qgsfeature.cpp#L68

I am no expert, but I would like to avoid having setters being executed too many times. It makes the debugging more difficult and might be noticeable by the user (rendering time, flickering).
but that's not my call to make here.

**Comment by nirvn** [comment]
BTW, animations also work, can make for nice attention grabbing elements, e.g.:
Peek 2020-08-17 15-35

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-3795451a96be56fb1ef9544c1a516e6ac44aea6f-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-3795451a96be56fb1ef9544c1a516e6ac44aea6f-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-3795451a96be56fb1ef9544c1a516e6ac44aea6f-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-3795451a96be56fb1ef9544c1a516e6ac44aea6f-x86_64.apk)

File: src/core/featuremodel.cpp
Comment (nirvn) [reply src/core/featuremodel.cpp:52]: I'll adjust.

Comment (nirvn) [reply src/core/featuremodel.cpp]: @3nids , done.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-7a6e7bdb052d7f3d3566b4647aef25061ea6260d-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-7a6e7bdb052d7f3d3566b4647aef25061ea6260d-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-7a6e7bdb052d7f3d3566b4647aef25061ea6260d-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-7a6e7bdb052d7f3d3566b4647aef25061ea6260d-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-4443e404c888799ec8b80d8f45cd74216909bf92-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-4443e404c888799ec8b80d8f45cd74216909bf92-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-4443e404c888799ec8b80d8f45cd74216909bf92-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-4443e404c888799ec8b80d8f45cd74216909bf92-x86_64.apk)

**Comment by nirvn** [comment]
HTML widget support added:
Peek 2020-08-18 11-58

File: src/core/attributeformmodelbase.cpp
Comment (3nids) [inline src/core/attributeformmodelbase.cpp:283]: remove?

Comment (nirvn) [reply src/core/attributeformmodelbase.cpp:283]: Thanks -- there's another debug line I'm removing too (the console.log in FeatureForm.qml)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-c3c86ccf384307b0b91a829a19e32d442447645f-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-c3c86ccf384307b0b91a829a19e32d442447645f-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-c3c86ccf384307b0b91a829a19e32d442447645f-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-c3c86ccf384307b0b91a829a19e32d442447645f-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-febd423837e2c2e262498273ce0c9e0932780c61-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-febd423837e2c2e262498273ce0c9e0932780c61-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-febd423837e2c2e262498273ce0c9e0932780c61-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-febd423837e2c2e262498273ce0c9e0932780c61-x86_64.apk)

**Comment by nirvn** [comment]
@m-kuhn , I think we might have a bit of work to do on the osgeo4a side of things to add the QML WebView module.

**Comment by nirvn** [comment]
@m-kuhn , I suspect a similar update will be required for QML charts module too. See how the QML chart widget is missing when testing on my smartphone:
image

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-447355c1e580b8204a1aed8e05f72b9654dd6f42-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-447355c1e580b8204a1aed8e05f72b9654dd6f42-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-447355c1e580b8204a1aed8e05f72b9654dd6f42-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-447355c1e580b8204a1aed8e05f72b9654dd6f42-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-de0e35dd12ce34de03f18da0222ead1205002f3c-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-de0e35dd12ce34de03f18da0222ead1205002f3c-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-de0e35dd12ce34de03f18da0222ead1205002f3c-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-de0e35dd12ce34de03f18da0222ead1205002f3c-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-9f3feea15f4029e8de84114e80945ebe2633076d-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-9f3feea15f4029e8de84114e80945ebe2633076d-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-9f3feea15f4029e8de84114e80945ebe2633076d-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-9f3feea15f4029e8de84114e80945ebe2633076d-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-04d7c32c3c4d323849a924975797b0a8ef0c3b95-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-04d7c32c3c4d323849a924975797b0a8ef0c3b95-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-04d7c32c3c4d323849a924975797b0a8ef0c3b95-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-04d7c32c3c4d323849a924975797b0a8ef0c3b95-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-51fe7a0bfdeb142f9c0e2006dfd8255415ab0320-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-51fe7a0bfdeb142f9c0e2006dfd8255415ab0320-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-51fe7a0bfdeb142f9c0e2006dfd8255415ab0320-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-51fe7a0bfdeb142f9c0e2006dfd8255415ab0320-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-53436dda03dc21951c48369d1ff290f1e6632fb7-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-53436dda03dc21951c48369d1ff290f1e6632fb7-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-53436dda03dc21951c48369d1ff290f1e6632fb7-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-53436dda03dc21951c48369d1ff290f1e6632fb7-x86_64.apk)

**Comment by nirvn** [comment]
Alright, HTML working on mobile :tada: 
image

@m-kuhn , charts still don't show up on mobile, are we missing some additional QML charts module?

**Comment by nirvn** [comment]
@m-kuhn , oh also, we need to address this:

```
  -- Configuring incomplete, errors occurred!
    Could not find a package configuration file provided by "Qt5WebView" with
    any of the following names:
  See also "/usr/src/build/CMakeFiles/CMakeOutput.log".
  
      Qt5WebViewConfig.cmake
      qt5webview-config.cmake
  
    Add the installation prefix of "Qt5WebView" to CMAKE_PREFIX_PATH or set
    "Qt5WebView_DIR" to a directory containing one of the above files.  If
    "Qt5WebView" provides a separate development package or SDK, be sure it has
    been installed.
```

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-921c4a80472c1b109f8f10b143e3953ce2568f72-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-921c4a80472c1b109f8f10b143e3953ce2568f72-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-921c4a80472c1b109f8f10b143e3953ce2568f72-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-921c4a80472c1b109f8f10b143e3953ce2568f72-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-e5abe827a2663e7bad175323cb7ff0825fcebc31-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-e5abe827a2663e7bad175323cb7ff0825fcebc31-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-e5abe827a2663e7bad175323cb7ff0825fcebc31-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-e5abe827a2663e7bad175323cb7ff0825fcebc31-x86_64.apk)

**Comment by m-kuhn** [comment]
Add WebView here https://github.com/opengisch/QField/blob/qml/src/core/CMakeLists.txt#L169

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-b1f60ca071ed77dd2faed073c03cae8b26043fda-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-b1f60ca071ed77dd2faed073c03cae8b26043fda-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-b1f60ca071ed77dd2faed073c03cae8b26043fda-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-b1f60ca071ed77dd2faed073c03cae8b26043fda-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-7978e96f789e30476fac031bff953e21e14089f2-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-7978e96f789e30476fac031bff953e21e14089f2-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-7978e96f789e30476fac031bff953e21e14089f2-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-7978e96f789e30476fac031bff953e21e14089f2-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-8eef8da65ac044d186b07c570c497848c2498844-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-8eef8da65ac044d186b07c570c497848c2498844-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-8eef8da65ac044d186b07c570c497848c2498844-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-8eef8da65ac044d186b07c570c497848c2498844-x86_64.apk)

**Comment by nirvn** [comment]
@m-kuhn , this is all green and ready to approve.

File: sdk.conf
Comment (m-kuhn) [inline sdk.conf:1]: Will need a release here before approval

File: src/core/attributeformmodel.h
Comment (m-kuhn) [inline src/core/attributeformmodel.h:55]: Could you add a docstring for me here?

File: src/qml/FeatureForm.qml
Comment (m-kuhn) [inline src/qml/FeatureForm.qml:375]: why are they not available ;-P

File: src/core/featuremodel.h
Comment (m-kuhn) [inline src/core/featuremodel.h:202]: Can you add a docstring here too?

File: src/core/featuremodel.cpp
Comment (m-kuhn) [inline src/core/featuremodel.cpp:351]: In most scenarios it's

```
if ( new != old ) 
{
  update();
  emitChanged();
}
```
what's different here?

Comment (nirvn) [reply src/core/featuremodel.cpp:351]: OK, so most of the time, "new" (i.e. the fetched feature from the provider after having updated said feature) will be equal to "old" ( the updated feature sent to the provider). For a small subset of feature updates, those which the provider subsequently modify on update, we end up calling setFeature( modifiedFeature[FromProvider] ) which emits a featureChanged signal.

But in most cases, when new == old, we had no way to connect to a signal to be aware of the feature having been updated. Hence the new featureUpdated signal.

File: src/qml/FeatureForm.qml
Comment (nirvn) [reply src/qml/FeatureForm.qml:375]: That's a pre-existing comment, I've not looking into this mysterious TODO :)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-9b06e1163879e10faae883957bcf423541ee5cbe-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-9b06e1163879e10faae883957bcf423541ee5cbe-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-9b06e1163879e10faae883957bcf423541ee5cbe-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-9b06e1163879e10faae883957bcf423541ee5cbe-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-2befbdea1ce6d15b1574cb77cbc97091e20161d5-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-2befbdea1ce6d15b1574cb77cbc97091e20161d5-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-2befbdea1ce6d15b1574cb77cbc97091e20161d5-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-2befbdea1ce6d15b1574cb77cbc97091e20161d5-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-ff43c33f648fd07524b688ce016b9bfaedc1cb7f-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-ff43c33f648fd07524b688ce016b9bfaedc1cb7f-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-ff43c33f648fd07524b688ce016b9bfaedc1cb7f-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-ff43c33f648fd07524b688ce016b9bfaedc1cb7f-x86_64.apk)

**Review by m-kuhn — APPROVED** [review]

File: .docker/testing/Dockerfile
Comment (m-kuhn) [inline .docker/testing/Dockerfile:5]: Can we put that to osgeo4a?

Comment (m-kuhn) [reply .docker/testing/Dockerfile:5]: both actually (will shorten build time)

Comment (nirvn) [reply .docker/testing/Dockerfile:5]: It's actually not needed. I was trying to see if that would bring the QML charts into the APK, it did not so I'm planning to revert.

Comment (m-kuhn) [reply .docker/testing/Dockerfile:5]: ah yeah, this one is pure linux tests, nothing related to apk

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-aaa6e9474398800a91e355341da114120e7e8c85-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-aaa6e9474398800a91e355341da114120e7e8c85-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-aaa6e9474398800a91e355341da114120e7e8c85-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1221-aaa6e9474398800a91e355341da114120e7e8c85-x86_64.apk)

**Pull request merged**
